AWSCognitoIdentityProvider/AWSCognitoIdentityUser.m

```diff
@@ -376,17 +377,22 @@ -(instancetype) initWithUsername: (NSString *)username pool:(AWSCognitoIdentityU
                 [requiredAttributesSet addObject:strippedKey];
             }
         }
-        AWSCognitoIdentityNewPasswordRequiredInput *newPasswordRequiredInput = [[AWSCognitoIdentityNewPasswordRequiredInput alloc] initWithUserAttributes:userAttributesDict requiredAttributes:requiredAttributesSet];
+        AWSCognitoIdentityNewPasswordRequiredInput *newPasswordRequiredInput = [[AWSCognitoIdentityNewPasswordRequiredInput alloc] initWithUserAttributes:userAttributesDict
+                                                                                                                                       requiredAttributes:requiredAttributesSet];
         AWSTaskCompletionSource<AWSCognitoIdentityNewPasswordRequiredDetails *> *newPasswordRequiredDetails = [AWSTaskCompletionSource<AWSCognitoIdentityNewPasswordRequiredDetails *> new];
         [newPasswordRequiredDelegate getNewPasswordDetails:newPasswordRequiredInput newPasswordRequiredCompletionSource:newPasswordRequiredDetails];
+        if (lastError) {
+            [newPasswordRequiredDelegate didCompleteNewPasswordStepWithError:lastError];
+        }
         return [[newPasswordRequiredDetails.task continueWithSuccessBlock:^id _Nullable(AWSTask<AWSCognitoIdentityNewPasswordRequiredDetails *> * _Nonnull task) {
             return [self performRespondToNewPasswordChallenge:task.result session:lastChallenge.session];
         }] continueWithBlock:^id _Nullable(AWSTask<AWSCognitoIdentityProviderRespondToAuthChallengeResponse *> * _Nonnull task) {
-            [newPasswordRequiredDelegate didCompleteNewPasswordStepWithError:task.error];
```

Comment:
This will break the [AWSMobileClient logic](https://github.com/aws-amplify/aws-sdk-ios/blob/master/AWSAuthSDK/Sources/AWSMobileClient/AWSMobileClientExtensions.swift#L872) which uses this delegate callback to return an error back to the user.

Also this will break the existing users of the old Cognito SDK.

Reply:
Thanks so much for bringing this up! You are indeed correct that it would have broken customers.  Thanks again.

I've come up with different approach which leaves the (old) view controller, on the navigation stack until the UIAlertView goes away.  I don't think this is any worse than what we are currently doing since any existing customers depending on existing functionality would have already had to work around the fact that we are pushing a view controller on the stack while previously calling `didCompleteNewPasswordStepWithError:`.  That being said, happy to know what you think about this approach.

Thanks!